peer.go

```diff
@@ -1087,142 +1178,162 @@ func (p *peer) handleInitClosingSigned(req *htlcswitch.ChanClose, msg *lnwire.Cl
 		IsPending:      true,
 	}
 	if err := channel.DeleteState(closeSummary); err != nil {
-		req.Err <- err
-		return
+		if localReq != nil {
+			localReq.Err <- err
+		}
+		return nil, 0
 	}
 
-	// Update the caller with a new event detailing the current pending
-	// state of this request.
-	req.Updates <- &lnrpc.CloseStatusUpdate{
-		Update: &lnrpc.CloseStatusUpdate_ClosePending{
-			ClosePending: &lnrpc.PendingUpdate{
-				Txid: closingTxid[:],
+	// If this is a locally requested shutdown, update the caller with a new
+	// event detailing the current pending state of this request.
+	if localReq != nil {
+		localReq.Updates <- &lnrpc.CloseStatusUpdate{
+			Update: &lnrpc.CloseStatusUpdate_ClosePending{
+				ClosePending: &lnrpc.PendingUpdate{
+					Txid: closingTxid[:],
+				},
 			},
-		},
+		}
 	}
 
 	_, bestHeight, err := p.server.cc.chainIO.GetBestBlock()
 	if err != nil {
-		req.Err <- err
-		return
+		if localReq != nil {
+			localReq.Err <- err
+		}
+		return nil, 0
 	}
 
 	// Finally, launch a goroutine which will request to be notified by the
 	// ChainNotifier once the closure transaction obtains a single
 	// confirmation.
 	notifier := p.server.cc.chainNotifier
-	go waitForChanToClose(uint32(bestHeight), notifier, req.Err,
-		req.ChanPoint, &closingTxid, func() {
+	var errChan chan error
```

Comment:
waitForChanToClose actually handles a `nil` error channel, and is currently called with the error chan set to nil in `master`. I see your point though, maybe we should rather start a goroutine to read the err chan in case of `localReq == nil` and print it?